docs/how-to-guides/deploy.md

```diff
@@ -0,0 +1,279 @@
+wish to replace the builder flavour (`m4.medium`) with `m2.medium` as there are
+generally more available.
+
+In `terraform/variables.tf`, you'll want to make the following changes:
+
+Change the default value of the `name` variable from "softpack" to
+"softpack-USERNAME" (where USERNAME is your username).
+
+Change the default value of `subdomain` to match that of `name`.
+
+Add a default value for `ANSIBLE_INVENTORY` set to `"./ansible/hosts.yml"`.
+
+Add a default value for `SSH_KEY` set to `"~/.ssh/id_rsa"`.
+
+NB: You may need to replace the default value of SSH_KEY with the path to your
+own keys if it is in a different location, or is of a different key type. If you
+do not already have SSH keys created, you can do so with the `ssh-keygen`
+command.
+
+Add a default value for `ENVIRONMENT` set to `"development"`.
+
+Add the following to the bottom of the file:
+
+```
+variable "SSH_PUBLIC_KEY" {
+  type = string
+  default = "~/.ssh/id_rsa.pub"
+}
+
+variable "openstack_auth_url" {
+  type = string
+  default = "https://theta.internal.sanger.ac.uk:5000"
+}
+
+variable "openstack_application_credential_id" {
+  type = string
+  default = "YOUR_CREDENTIAL_SECRET"
+}
+
+variable "openstack_application_credential_secret" {
+  type = string
+  default = "YOUR_CREDENTIAL_SECRET"
+}
+```
+
+…filling in your credentials that you generated earlier, and making sure the
+`SSH_PUBLIC_KEY` path matches that of the accompanying public key to the
+`SSH_KEY` you supplied earlier.
+
+Lastly, in the `terraform/ansible/roles/softpack_builder/tasks/main.yml` file,
+**remove** the following lines:
+
+```
+- name: Import GPG key
+  vars:
+    key: /tmp/softpack.pgp
+  block:
+    - name: Copy GPG key
+      ansible.builtin.copy:
+        src: "{{ lookup('ansible.builtin.env', 'SPACK_KEY') }}"
+        dest: "{{ key }}"
+        mode: u=rw,g=-,o=-
+
+    - name: Import GPG key
+      ansible.builtin.shell: |
+        SPACK_GNUPGHOME={{ builder.gpg }} {{ spack_path }}/bin/spack gpg trust {{ key }}
+      args:
+        creates: "{{ builder.gpg }}/trustdb.gpg"
+```
+
+## Building the VM
+
+For this stage, you'll need to have both [Terraform](https://www.terraform.io/)
+and [Ansible](https://www.ansible.com/) installed. It is recommended to install
+`ansible` using `pip`:
+
+```
+pip install ansible
+```
+
+Once done, make sure that you're in the `infrastructure/terraform` directory,
+and run the following two commands to provision the VM for SoftPack:
+
+```
+terraform init
+terraform apply
+```
+
+…following the onscreen instructions to confirm the actions that will take
+place.
+
+## Installing the Software
+
+While still in the `infrastructure/terraform` directory, you can run the
+following command to set-up the newly provisioned VMs with the necessary
+software and configuration:
+
+```
+ansible-playbook --inventory ansible/hosts.yml ansible/main.yml
+```
+
+You will need to response 'yes' to each of the three SSH authenticity checks,
```

Comment:
It didn't ask me anything, but seemed to complete fine. Should I be concerned?...

Reply:
If you already SSH'd into the VMs then it won't ask, and there's probably an SSH config option to disable the question. If you didn't already SSH in, and don't have that config option set, then I don't know why it didn't ask.

Reply:
I don't think I'd already ssh'd in, so we'll just consider this a mystery!